### PR TITLE
[4.0] Template Style assignment: Fix toggle checkboxes

### PR DESCRIPTION
**Pull request by infograf768**
Replaces https://github.com/joomla/joomla-cms/pull/24335
Thanks @bahl24
### Summary of Changes
Creates an es6.js file in the build/media_source/com_templates/

### Testing Instructions
Patch.
Run npm
Edit Cassiopea template style
Template(Styles) -> Menu Assignment -> 'Toggle Selection' buttons

### After patch
template_assignment

### Note: 
There may be another way to format the es6.js. Please comment before merge.
@C-Lodder
@dgrammatiko

### CODE REVIEW

**Comment by infograf768** [comment]
I guess I need help to solve hound : `is defined but never used no-unused-vars`...

**Comment by dgrammatiko** [comment]
@infograf768 couple things here:
- DO NOT use Global vars (eg instead of `toggleAll()` you need to do `Joomla.toggleAll()` )
- Wrap the code in an IIFE ( `() => { /** your code here */ }();`

eg:
```js
/**
 * @copyright  Copyright (C) 2005 - 2019 Open Source Matters, Inc. All rights reserved.
 * @license    GNU General Public License version 2 or later; see LICENSE.txt
 */
Joomla = window.Joomla || {};

(Joomla => {
  Joomla.toggleAll = () => {
    const checkBoxes = [].slice.call(document.querySelectorAll('.chk-menulink'));
    const value = checkBoxes[0].checked;
    checkBoxes.forEach((checkBox) => {
      checkBox.checked = !value;
    });
  };

  Joomla.toggleMenutype = (a) => {
    const checkBox = [].slice.call(document.getElementsByClassName(a));
    const value = checkBox[0].checked;
    checkBox.forEach((element) => {
      element.checked = !value;
    });
  };
})(Joomla);

```

**Comment by infograf768** [comment]
Tks, testing now

File: build/media_source/com_templates/js/admin-template-toggle-assignment.es6.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * @license    GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+/**
+ * @copyright  Copyright (C) 2005 - 2019 Open Source Matters, Inc. All rights reserved.
+ * @license    GNU General Public License version 2 or later; see LICENSE.txt
+ */
+Joomla = window.Joomla || {};
+
+(Joomla => {
```
[the remark this answers is not in the document]
Comment (dgrammatiko) [reply build/media_source/com_templates/js/admin-template-toggle-assignment.es6.js]: `((Joomla) => {` my bad

[the remark this answers is not in the document]
Comment (infograf768) [reply build/media_source/com_templates/js/admin-template-toggle-assignment.es6.js]: Yep, already saw it and corrected.
Thanks!

**Comment by infograf768** [comment]
Would be nice to use hound on localhost

**Comment by roland-d** [comment]
@infograf768 It is called Code Sniffer. See https://docs.joomla.org/Joomla_CodeSniffer

**Comment by dgrammatiko** [comment]
@infograf768 you can use the lint command eg `npx eslint .` or `npm run lint:js`
Bonus you can run `npx eslint . --fix` to autofix the CS

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on ed17974a4d17a53c08ac55ff1b7fda6617939868<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25155">issues.joomla.org/tracker/joomla-cms/25155</a>.</sub>

File: build/media_source/com_templates/js/admin-template-toggle-assignment.es6.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/**
+ * @copyright  Copyright (C) 2005 - 2019 Open Source Matters, Inc. All rights reserved.
+ * @license    GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+/**
+ * @copyright  Copyright (C) 2005 - 2019 Open Source Matters, Inc. All rights reserved.
+ * @license    GNU General Public License version 2 or later; see LICENSE.txt
+ */
```
Comment (richard67) [inline build/media_source/com_templates/js/admin-template-toggle-assignment.es6.js]: Is that by purpose that we have 2 times same copyright information, lines1 to 4 and again lines 6 to 9?

**Comment by richard67** [comment]
I have not tested this item.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25155">issues.joomla.org/tracker/joomla-cms/25155</a>.</sub>

File: build/media_source/com_templates/js/admin-template-toggle-assignment.es6.js (hunk 2 above)
Comment (richard67) [reply build/media_source/com_templates/js/admin-template-toggle-assignment.es6.js]: I suggest to remove lines 5 to 9.

**Review by richard67 — CHANGES_REQUESTED** [review]

**Comment by infograf768** [comment]
@richard67 
Should be OK now.

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 37b980903d2c55ffa1414977e28dbb409c6ac25f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25155">issues.joomla.org/tracker/joomla-cms/25155</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 37b980903d2c55ffa1414977e28dbb409c6ac25f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25155">issues.joomla.org/tracker/joomla-cms/25155</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25155">issues.joomla.org/tracker/joomla-cms/25155</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**